data/model/oci/manifest.py

```diff
@@ -392,6 +321,121 @@ def _create_manifest(
     return CreatedManifest(manifest=manifest, newly_created=True, labels_to_apply=labels_to_apply)
 
 
+def _build_blob_map(
+    repository_id,
+    manifest_interface_instance,
+    retriever,
+    storage,
+    raise_on_error=False,
+    require_empty_layer=True,
+):
```

Comment:
Would you mind to include a docstring for this function? It has a lot of arguments and is relatively complex

Reply:
Done